### ISSUE TITLE
Behavior at the ends of recurrence sequences; or, what does "recur" mean?

### ISSUE THREAD

**Issue by randallsquared** [awinder/iso8601-navigate#2]
The term "recurrence" in the ISO8601 (2004) standard is ambiguous. A recurrence is indicated by a leading `R` and a trailing number with zero or more digits (`R/...` indicates unbounded recurrences).  The standard does not define what `R0/...` or `R1/...` mean, and there are no such examples listed.  In plain english, the terms "recurrence" or "repetition" sometimes mean every event after the first of a series of events, and sometimes refer to the series of events entire, including the initial event of which the remainder are the recurrences.
1. If we take "recurrence" to mean the complete count of events
   - then `R5/...` refers to a series of five events, `R1/...` to a single event, and `R0/...` is either an expression wherein the rest is ignored and no events are referenced, or has no meaning.
2. If we take "recurrence" to mean all events after the first
   - then `R5/...` may refer to the latter five events in an implicit series of six events, or
   - to the series of six events.
   
   This is still somewhat ambiguous, but at least in this case we have a meaning for `R0/...`, an expression which is not disallowed by the language of the standard, but otherwise has no meaning.

This problem extends to all values of R, not just zero or one, since we'll often want to ask what is the first event in our series, or call `.next()` or `.previous()` until we reach one or the other end of the series.

Assuming we had method on iso8601-navigator that returned all (known) events in a series, here are the possibilities for what navigator('R4/2012-07-01T00:00:00Z/P7D').all() should return:
1. The series of events includes the initial event of which the interval string is a recurrence:  
   `['2012-07-01T00:00:00Z', '2012-07-08T00:00:00Z', '2012-07-15T00:00:00Z', '2012-07-22T00:00:00Z', '2012-07-28T00:00:00Z']`
2. The initial occurrence is the specified start date, but recurrence events are the trailing edge of intervals:  
   `['2012-07-08T00:00:00Z', '2012-07-15T00:00:00Z', '2012-07-22T00:00:00Z', '2012-07-28T00:00:00Z']`
3. The initial occurrence is out of scope and exists previously to the recurrence start date:  
   `['2012-07-01T00:00:00Z', '2012-07-08T00:00:00Z', '2012-07-15T00:00:00Z', '2012-07-22T00:00:00Z']`
   - implies that `R0/...` is either invalid or refers to an empty list of recurrence events

Various libraries for parsing ISO8601 recurring intervals have handled this in different ways:  
- [Python's aniso8601](https://pypi.python.org/pypi/aniso8601/) takes
  interpretation 3 and converts `R0/...` to an empty list of events.  
- [PHP's DatePeriod](http://php.net/manual/en/dateperiod.construct.php) takes interpretation 1 by default, and has an option (`DatePeriod::EXCLUDE_START_DATE`) to behave as interpretation 2.

My feeling is that we should support 1 by default and allow both 2 and 3 as optional flags to the factory.

**Comment by inadarei** [comment]
o_O
